### PR TITLE
Add collapsible element to updater notification to show changelog, open if there breaking changes

### LINKED ISSUE #3984: Changelog view or Breaking Change Message
As we discussed in the post mortem, users had no visibility of the `circle` breaking change.  This could be through
- alert
- error message on the code that broke. (error message on `circle` that has an is-was once the user hovers over it)

@jgomez720 to create an email template for these breaking changes in the future in hubspot
https://github.com/KittyCAD/pet-store/issues/187

@jessfraz reassign to those you need

### PR DESCRIPTION
**Pull request by franknoirot**
Closes #3984. Need to test the updater to ensure that we still get the markdown of the release notes to be able to parse them. I think I still need to add error handling if the notes are undefined.

Yeah at least in the updater test I'm not receiving any release notes to parse, so I'm getting an error. Need to test the component out with some fake notes data (oh and I can write a component test for it!), then coordinate with @pierremtb to make sure the app has the release notes markdown when returned from the auto-updater.

### CODE REVIEW

**Comment by pierremtb** [comment]
@franknoirot I'm trying to push a dummy updater-test build with https://github.com/KittyCAD/modeling-app/tree/updater-test that would include the release notes so we can test this branch with that. They aren't currently being populated on release

**Comment by pierremtb** [comment]
Need to fix this windows failure before we can move on though

**Comment by franknoirot** [comment]
> Need to fix this windows failure https://github.com/KittyCAD/modeling-app/actions/runs/11214762223/job/31170760758 before we can move on though
> 
> New attempt is at https://github.com/KittyCAD/modeling-app/actions/runs/11214995573

Great, thank you. I'll still make the component resilient to if release notes aren't present, but that will be very helpful for confirming it's appearance.

**Comment by pierremtb** [comment]
Made good progress, had to rework the way notes were published I had understood the docs wrong. This is what it looks like in an updater test related to #4123 
```
Found version 0.255.255 (url: Zoo Modeling App-0.255.255-x64-mac.zip, Zoo Modeling App-0.255.255-arm64-mac.zip, Zoo Modeling App-0.255.255-x64-mac.dmg, Zoo Modeling App-0.255.255-arm64-mac.dmg)
update-available {
  version: '0.255.255',
  files: [
    {
      url: 'Zoo Modeling App-0.255.255-x64-mac.zip',
      sha512: 'VJb0qlrqNr+rVx3QLATz+B28dtHw3osQb5/+UUmQUIMuF9t0i8dTKOVL/2lyJSmLJVw2/SGDB4Ud6VlTPJ6oFw==',
      size: 141277345
    },
    {
      url: 'Zoo Modeling App-0.255.255-arm64-mac.zip',
      sha512: 'b+ugdg7A4LhYYJaFkPRxh1RvmGGMlPJJj7inkLg9PwRtCnR9ePMlktj2VRciXF1iLh59XW4bLc4dK1dFQHMULA==',
      size: 135278259
    },
    {
      url: 'Zoo Modeling App-0.255.255-x64-mac.dmg',
      sha512: 'gCUqww05yj8OYwPiTq6bo5GbkpngSbXGtenmDD7+kUm0UyVK8WD3dMAfQJtGNG5HY23aHCHe9myE2W4mbZGmiQ==',
      size: 146004232
    },
    {
      url: 'Zoo Modeling App-0.255.255-arm64-mac.dmg',
      sha512: 'ND871ayf81F1ZT+iWVLYTc2jdf/Py6KThuxX2QFWz14ebmIbJPL07lNtxQOexOFiuk0MwRhlCy1RzOSG1b9bmw==',
      size: 140021522
    }
  ],
  path: 'Zoo Modeling App-0.255.255-x64-mac.zip',
  sha512: 'VJb0qlrqNr+rVx3QLATz+B28dtHw3osQb5/+UUmQUIMuF9t0i8dTKOVL/2lyJSmLJVw2/SGDB4Ud6VlTPJ6oFw==',
  releaseNotes: 'Non-release build, commit d6488ecf784c0559bcbeaacc1fc4f22820037846\n',
  releaseDate: '2024-10-09T11:57:59.133Z'
}
```

**Comment by pierremtb** [comment]
Thanks @franknoirot for merging #3995! Just updated from main, hope I didn't mess up conflict resolution

**Comment by franknoirot** [comment]
Just finishing some component test finessing but here's what I'm thinking for the look for now

image

**Comment by franknoirot** [comment]
And if the word "breaking" appears in the release notes:

image

File: .github/workflows/build-test-publish-apps.yml [hunk 1]
```diff
@@ -13,8 +13,8 @@ on:
   # Will checkout the last commit from the default branch (main as of 2023-10-04)
 
 env:
-  CUT_RELEASE_PR: ${{ github.event_name == 'pull_request' && (contains(github.event.pull_request.title, 'Cut release v')) }}
-  BUILD_RELEASE: ${{ github.event_name == 'release' || github.event_name == 'schedule' || github.event_name == 'pull_request' && (contains(github.event.pull_request.title, 'Cut release v')) }}
+  CUT_RELEASE_PR: true
+  BUILD_RELEASE: true
```
Comment (franknoirot) [inline .github/workflows/build-test-publish-apps.yml:17]:
```suggestion
  CUT_RELEASE_PR: ${{ github.event_name == 'pull_request' && (contains(github.event.pull_request.title, 'Cut release v')) }}
  BUILD_RELEASE: ${{ github.event_name == 'release' || github.event_name == 'schedule' || github.event_name == 'pull_request' && (contains(github.event.pull_request.title, 'Cut release v')) }}
```

Don't let me merge this without reverting this part

**Comment by pierremtb** [comment]
🔥

**Comment by pierremtb** [comment]
updater-test looking good here, let me remove the release builds for merge!

image

**Review by pierremtb — APPROVED** [review]
LGTM!

**Pull request merged**